### PR TITLE
Read RRFS bkgd cld fraction for use in the cloud analysis

### PR DESCRIPTION
**Pull request by cshartsough**
## DESCRIPTION OF CHANGES:
Tests have shown that the RRFS cloud analysis can be improved by including a test in the cloud-building stage, to see how high the background cloud fraction is. If there is sufficient cloud fraction (e.g., >0.45) then no building is performed.

To add cloud fraction, it is necessary to include the links to the RRFS intermediate file (phy_data.nc) containing the cloud fraction values. This change is is PR #427.

This PR contains the Fortran code changes to use the cloud fraction data in the cloud analysis.

## TESTS CONDUCTED:
A 5-day retrospective case was run for Spring 2021, using the RRFS Base D version. Verification showed that there were small improvements in low-level moisture (RH) and in ceiling, especially the bias values.

## DEPENDENCIES:
This change depends on PR #427 (linking the RRFS cloud fraction data) to use cloud fraction in the cloud analysis.

## CONTRIBUTORS (optional):
Terra Ladwig

### CODE REVIEW

File: cloudanalysis/cloudanalysis_fv3driver.f90 [hunk 1]
```diff
@@ -796,6 +797,7 @@ program cloudanalysis
 !
 !  call MPI_Barrier(mpi_comm_world, ierror)
   do k=1,nsig
+     kk=nsig+1-k
```
Comment (cshartsough) [inline cloudanalysis/cloudanalysis_fv3driver.f90:808]: The phy_data file uses vertical levels from bottom to top, opposite of the other intermediate netcdf files and the expected direction in the Fortran code.  I use the kk index to reverse the direction; if there is a better way to do it I can make that change.

**Review by hu5970 — APPROVED** [review]
looks good

**Comment by hu5970** [comment]
@chunhuazhou @guoqing-noaa Please review this PR. Thanks, Ming

File: cloudanalysis/cloudanalysis_fv3driver.f90 (hunk 1 above)
Comment (christinaholtNOAA) [reply cloudanalysis/cloudanalysis_fv3driver.f90:808]: I'd say this GitHub comment would make a fantastic code comment.

**Review by christinaholtNOAA — APPROVED** [review]
This looks good to me. Just one comment below, if you don't mind addressing that before you merge.

File: cloudanalysis/cloudanalysis_fv3driver.f90 (hunk 1 above)
Comment (cshartsough) [reply cloudanalysis/cloudanalysis_fv3driver.f90:808]: thanks for the idea.  Done.

**Pull request merged**